### PR TITLE
Feature/marketplace-v3:  Add articles to user collection

### LINKED ISSUE #376: [Marketplace v3] Add articles to the user collection
* base branch: `feature/marketplace-v3`

- [x] create a new style of subtabs with the vertical separator (see design)
- [x] add subtabs to the user's collection
- [x] `/u/username/collection` & `/pkh/id/collection` points to the collection of gentks
- [x] `/u/username/collection/articles` & `/pkh/id/collection/articles` points to the collection of articles
- [x] the articles sub tab shows the list of articles owned by the user (same vertical layout as explore articles, no search/filter bar at the top, raw list))
- [x] move the `ENJOY COLLECTION` button in the header bar of the gentk collection (see design)

The list of articles owned by the user is exposed by the `articlesOwned` entry of the `user` entity (no pagination):

```gql
query User($userId: String) {
  user(id: $userId) {
    id
    name
    articlesOwned {
      amount
      article {
        id
        title
      }
    }
  }
}
```

Design: https://www.figma.com/file/bdCVYIEKTy8lyBLjIDWvQn/Website?node-id=35%3A1877
image

You can use this account for testing the feature as it owns many articles: `/pkh/tz1MGzgRu6qJ3RaBUErnpFDLarFVPgaApKrS`

### PR DESCRIPTION
**Pull request by maerzhase**
- fix #376

Implemented as described in #376. 

@ciphrd For now we are just fetching objkts and articles owned in parallel independent of the selected tab. what do you think?

Link: http://localhost:3000/u/fxhash%20admin/collection/articles

Screenshot 2022-09-08 at 17 29 59

Link: http://localhost:3000/u/fxhash%20admin/collection

Screenshot 2022-09-08 at 17 32 25

### CODE REVIEW

File: src/components/Layout/Tabs.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export type TabDefinition = {
   name: string,
   props?: any
 }
-type TabsLayout = "full-width" | "fixed-size" | "subtabs"
+type TabsLayout = "full-width" | "fixed-size" | "subtabs" | "subtabs-vertical"
```
Comment (maerzhase) [inline src/components/Layout/Tabs.tsx:21]: the new type `subtabs-vertical` creates the new style of tabs with a vertical separator

File: src/containers/User/UserProfileLayout.tsx [hunk 2]
```diff
@@ -22,12 +22,14 @@ const TabWrapper = ({ children, ...props }: TabWrapperProps) => (
 
 interface Props {
   user: User
-  activeTab: 'creations' | 'articles' | 'collection' | 'on-sale' | 'dashboard'
+  activeTab: 'creations' | 'articles' | 'collection' | 'on-sale' | 'dashboard',
+  hideSectionSpacing?: boolean
```
Comment (maerzhase) [inline src/containers/User/UserProfileLayout.tsx:26]: Added this boolean for hiding the spacing above the section content. This was needed to move the `subtabs-vertical` closer to the main tab navigation

File: src/components/Button/index.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   size?: ButtonSize
   color?: "black" | "primary" | "secondary" | "transparent"
   className?: string
-  iconSide?: "left" | "right"
+  iconSide?: "left" | "right" | null
```
[the remark this answers is not in the document]
Comment (ciphrd) [reply src/components/Button/index.tsx:18]: LGTM

File: src/components/Layout/Tabs.tsx (hunk 1 above)
Comment (ciphrd) [reply src/components/Layout/Tabs.tsx:21]: 👍

File: src/containers/User/UserProfileLayout.tsx (hunk 2 above)
Comment (ciphrd) [reply src/containers/User/UserProfileLayout.tsx:26]: LGTM

**Review by ciphrd — APPROVED** [review]
I've made a few changes, mainly some refacto to move each section into sub components (clearer + saves a query)
Also went over a few design details.

**Comment by maerzhase** [comment]
Jeah looks nice! thats what i was wondering to do. don't hesitate to assign those refactors to me as well.

**Pull request merged**